docs/modules/ROOT/pages/cookbook.adoc

```diff
@@ -964,3 +964,179 @@ payload + { "middleName": "NMN",
   "title": "Duke of Java"
 }
 -------------
+
+### Reading and writing XML
+
+DataSonnet supports XML as an input and output format. xref:headers.adoc[Headers] can be used to control the behavior of the mapper.
+
+#### Reading XML
+Let's take the following XML input as an example:
+
+--------------
+<?xml version="1.0" encoding="UTF-8"?>
+<test:root xmlns:test="http://www.datasonnet.com">
+    <test:datasonnet version="1.0">Hello World</test:datasonnet>
+</test:root>
+--------------
+
+The internal representation of this input in DataSonnet would be:
+
+--------------
+{
+  "test:root": {
+    "@xmlns": {
+      "test": "http://www.datasonnet.com"
+    },
+    "test:datasonnet": {
+      "@version": "1.0",
+      "$": "Hello World"
+    }
+  }
+}
+--------------
+
+Our mapping can simply extract the value of the `<test:datasonnet>` elemeng, e.g.:
+
+--------------
+{
+  greeting: payload["test:root"]["test:datasonnet"]["$"]
+}
+--------------
+
+Note that HashMap-like notation is used to access the value, because colon `:` cannot be used in the DataSonnet identifier. In addition, the text value of the element is mapped to a dollar sign `$` which is not a valid identifier either. Let's use headers to fix this:
```

Comment:
`HashMap-like` is probably confusing to most people reading this. Maybe refer to the notation directly? "Note that `["key" ]` syntax is used to look up the property, not `.key` syntax, because..." ?

Reply:
good point, fixed